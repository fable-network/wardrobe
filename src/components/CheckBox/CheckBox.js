import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled('span')`
  pointer-events: ${p => (p.disabled ? 'none' : 'initial')};
  opacity: ${p => (p.disabled ? '0.3' : 'initial')};
  cursor: pointer;
  line-height: 1;
  display: inline-flex;
  align-items: center;
`;

const Input = styled('span')`
  display: inline-block;
  position: relative;
  width: 0.8125em;
  height: 0.8125em;
  border-radius: 50%;
  box-shadow: 0px 0px 0px 1px ${p => (p.checked ? p.theme.primary : p.theme.dark)};
  background: ${p => (p.checked ? p.theme.primary : 'transparent')};
  border: solid 0.05em #fff;
  box-sizing: border-box;
  transition: background 0.3s ease-out;
`;

const Label = styled('span')`
  display: inline-block;
  margin-left: 8px;
  user-select: none;
`;

const InputWrapper = styled('span')`
  display: flex;
  align-items: center;
`;

const CheckBox = ({ label, checked, disabled, onToggle, ...otherProps }) => (
  <Wrapper {...otherProps} onClick={onToggle} disabled={disabled}>
    <InputWrapper>
      <Input checked={checked} />
    </InputWrapper>
    <Label>{label}</Label>
  </Wrapper>
);

CheckBox.defaultProps = {
  onToggle: () => null,
};

CheckBox.propTypes = {
  onToggle: PropTypes.func,
  label: PropTypes.string,
  disabled: PropTypes.bool,
  checked: PropTypes.bool,
};

export default CheckBox;
