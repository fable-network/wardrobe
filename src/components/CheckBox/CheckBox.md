Unchecked:
```jsx
<CheckBox label="Checkbox Label" />
```

Checked:
```jsx
<CheckBox label="Checkbox Label" isChecked />
```

Disabled:
```jsx
<CheckBox label="Checkbox Label" isDisabled />
```

Interactable example:
```jsx
class CheckBoxExample extends React.Component {
  constructor() {
    super();
    this.handleToggle = this.handleToggle.bind(this);
    this.state = { isChecked: false };
  }

  handleToggle() {
    this.setState(({ isChecked }) => ({ isChecked: !isChecked }));
  }

  render() {
    const { isChecked } = this.state;
    const label = isChecked ? "Yaay I'm selected" : "Select me";

    return (
      <CheckBox label={label} onToggle={this.handleToggle} isChecked={isChecked}/>
    );
  }
}
<CheckBoxExample />
```
