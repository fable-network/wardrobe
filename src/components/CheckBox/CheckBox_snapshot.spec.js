/* eslint-disable */
// This file was automatically generated
import React from 'react';
import { shallow } from 'enzyme';
import CheckBox from './CheckBox';

describe('CheckBox Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = shallow(
        <CheckBox label="Checkbox Label" />
      );
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = shallow(
        <CheckBox label="Checkbox Label" isChecked />
      );
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = shallow(
        <CheckBox label="Checkbox Label" isDisabled />
      );
      expect(wrapper3).toMatchSnapshot();

    });
  });
});
