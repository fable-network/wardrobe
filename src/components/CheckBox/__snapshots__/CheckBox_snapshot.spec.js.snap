// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBox
    label="Checkbox Label"
    onToggle={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CheckBox__InputWrapper>
          <CheckBox__Input />
        </CheckBox__InputWrapper>,
        <CheckBox__Label>
          Checkbox Label
        </CheckBox__Label>,
      ],
      "isDisabled": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <CheckBox__Input />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isChecked": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Checkbox Label",
        },
        "ref": null,
        "rendered": "Checkbox Label",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CheckBox__InputWrapper>
            <CheckBox__Input />
          </CheckBox__InputWrapper>,
          <CheckBox__Label>
            Checkbox Label
          </CheckBox__Label>,
        ],
        "isDisabled": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CheckBox__Input />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isChecked": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Checkbox Label",
          },
          "ref": null,
          "rendered": "Checkbox Label",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`CheckBox Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBox
    isChecked={true}
    label="Checkbox Label"
    onToggle={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CheckBox__InputWrapper>
          <CheckBox__Input
            isChecked={true}
          />
        </CheckBox__InputWrapper>,
        <CheckBox__Label>
          Checkbox Label
        </CheckBox__Label>,
      ],
      "isDisabled": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <CheckBox__Input
            isChecked={true}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isChecked": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Checkbox Label",
        },
        "ref": null,
        "rendered": "Checkbox Label",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CheckBox__InputWrapper>
            <CheckBox__Input
              isChecked={true}
            />
          </CheckBox__InputWrapper>,
          <CheckBox__Label>
            Checkbox Label
          </CheckBox__Label>,
        ],
        "isDisabled": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CheckBox__Input
              isChecked={true}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isChecked": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Checkbox Label",
          },
          "ref": null,
          "rendered": "Checkbox Label",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`CheckBox Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBox
    isDisabled={true}
    label="Checkbox Label"
    onToggle={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CheckBox__InputWrapper>
          <CheckBox__Input />
        </CheckBox__InputWrapper>,
        <CheckBox__Label>
          Checkbox Label
        </CheckBox__Label>,
      ],
      "isDisabled": true,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <CheckBox__Input />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isChecked": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Checkbox Label",
        },
        "ref": null,
        "rendered": "Checkbox Label",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CheckBox__InputWrapper>
            <CheckBox__Input />
          </CheckBox__InputWrapper>,
          <CheckBox__Label>
            Checkbox Label
          </CheckBox__Label>,
        ],
        "isDisabled": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <CheckBox__Input />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isChecked": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Checkbox Label",
          },
          "ref": null,
          "rendered": "Checkbox Label",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
